src/docs/cursor-ops.dox

```diff
@@ -161,4 +161,52 @@ that may not be modified or freed by the application.  If a longer scope
 is required, the application must make a copy of the memory before the
 cursor is re-used, closed or reset.
 
+@section cursor_bound Apply bounds to the cursor
+
+A cursor may have a set of bounds applied to it, this restricts the key range
+that the cursor operates within.
+
+Suppose we have a table with keys A to Z. We can restrict the cursor to only operate over
+keys L->Q. If the cursor walks past key Q, the cursor will enter an un-positioned state having
+walked off the "end" of the table from it's perspective. Additionally should the cursor perform a
+search_near it will only search within the key range of L->Q. This can provide a performance boost
+when operating within tables that have a large number of deleted or invisible records. The cursor
+can have a lower bound set or an upper bound set, or both.
```

Comment:
I'd add a little more on the performance note:  This can provide **an application with** a performance boost
when operating within tables that have a large number of deleted or invisible records **just outside the boundaries**.

Reply:
True, the deleted records can exist within the boundaries, but does the app get a performance benefit over regular cursor traversal?  Assume boundaries at 10 and 20 and there are visible values at 9, 13, 15, 21 with deletions at all numbers in between 11,12,14,16,17,18,19.  Isn't the performance the same as not using boundaries?  I'm sure I'm missing something - if there's something more subtle maybe that's something that could be in the doc.

Reply:
I think you're not missing anything I just think I wanted to be explicit about why you'd get a performance speedup and that's because we check the key before we do any visibility checking. The performance would only optimize cases where the deleted keys are on or after the bounds.

Reply:
Yes, makes sense.

Reply:
How about: 
```
This can provide an application with a performance boost when operating within tables that have a large number of deleted or invisible records. This is because the key check occurs before we determine if the value associated with the key is visible or not.
```
That seems like the best of both worlds.

Reply:
Great!